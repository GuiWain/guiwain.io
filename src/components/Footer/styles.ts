import styled, { css } from 'styled-components'

import { Language } from '@styled-icons/ionicons-sharp/'
import { Aboutdotme } from '@styled-icons/simple-icons/'
import { FilterAlt } from '@styled-icons/boxicons-regular/'
import { ToTop } from '@styled-icons/boxicons-solid/'
import { Github } from '@styled-icons/entypo-social/'

export const LanguageIcon = styled(Language)``
export const AboutIcon = styled(Aboutdotme)``
export const FilterIcon = styled(FilterAlt)``
export const ArrowIcon = styled(ToTop)``

export const GithubIcon = styled(Github)`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    transition: color ${theme.transition.default};

    width: 3rem;
    height: 3rem;

    cursor: pointer;

    :hover {
      color: ${theme.colors.secondaryVariation};
    }
  `}
`

export const Wrapper = styled.div`
  width: 100%;
  margin-top: 7rem;
  background-color: transparent;

  display: flex;
  flex-direction: column;
`

export const Address = styled.address`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 0.1rem;

  margin-bottom: 7rem;
`

export const Author = styled.span`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-family: ${theme.font.family02};
    font-size: ${theme.font.sizes.xxxsmall};
    font-weight: ${theme.font.light};
  `}
`

export const Year = styled.time`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-family: ${theme.font.family02};
    font-size: ${theme.font.sizes.xxxsmall};
    font-weight: ${theme.font.light};
  `}
`

export const Anchor = styled.a``

export const Footer = styled.footer`
  ${({ theme }) => css`
    background: ${theme.colors.secondaryBg};
    box-shadow: 0px 0px 1px 0px rgba(122, 0, 65, 0.3);
    transition: background ${theme.transition.default},
      box-shadow ${theme.transition.default};

    width: 100%;
    height: 5rem;
    margin-top: 2rem;

    position: fixed;
    bottom: 0;
    left: 0;

    :hover {
      box-shadow: 0px -1px 3px 0px rgba(122, 0, 65, 0.2);
    }
  `}
`

export const Nav = styled.nav`
  ${({ theme }) => css`
    width: 90%;
    height: 100%;
    margin: auto;

    display: flex;
    justify-content: space-between;
    align-items: center;

    ${LanguageIcon},
    ${AboutIcon},
    ${FilterIcon},
    ${ArrowIcon} {
      color: ${theme.colors.icons};
      transition: color ${theme.transition.default};

      width: auto;
      height: 3rem;

      cursor: pointer;

      :hover {
        color: ${theme.colors.secondaryVariation};
      }
    }
  `}
`
