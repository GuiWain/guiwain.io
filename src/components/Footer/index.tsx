import * as S from './styles'

const Footer = () => (
  <S.Wrapper>
    <S.Nav>
      <S.LanguageIcon />
      <S.AboutIcon />
      <S.FilterIcon />
      <S.ArrowIcon />
    </S.Nav>
  </S.Wrapper>
)

export default Footer
