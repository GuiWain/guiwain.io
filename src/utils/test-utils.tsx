import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'
import * as theme from 'styles/theme'

const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme.light}>{children}</ThemeProvider>)

export * from '@testing-library/react'
export { renderWithTheme as render }
