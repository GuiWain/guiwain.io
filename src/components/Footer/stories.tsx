import { ComponentMeta, ComponentStory } from '@storybook/react'
import Footer from '.'

export default {
  title: 'Components/Footer',
  component: Footer
} as ComponentMeta<typeof Footer>

export const Default: ComponentStory<typeof Footer> = (args) => (
  <Footer {...args} />
)

Default.args = {
  modifier: 'default'
}

export const Minimal: ComponentStory<typeof Footer> = (args) => (
  <Footer {...args} />
)

Minimal.args = {
  modifier: 'minimal'
}
