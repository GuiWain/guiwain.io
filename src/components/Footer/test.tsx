import { render } from 'utils/test-utils'

import Footer from '.'

describe('<Footer />', () => {
  it('should render the Footer', () => {
    const { container } = render(<Footer />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
