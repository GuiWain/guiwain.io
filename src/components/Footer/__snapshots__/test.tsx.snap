// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render the Footer 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  background: #ADADAD;
  box-shadow: 0px 0px 1px 0px rgba(122,0,65,0.3);
  -webkit-transition: background 0.4s ease-in-out,box-shadow 0.4s ease-in-out;
  transition: background 0.4s ease-in-out,box-shadow 0.4s ease-in-out;
  width: 100%;
  height: 5rem;
  position: absolute;
  bottom: 0;
  left: 0;
}

.c0:hover {
  box-shadow: 0px -1px 3px 0px rgba(122,0,65,0.2);
}

.c1 {
  width: 90%;
  height: 100%;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0"
>
  <nav
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="c2 "
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M363 176L246 464h47.24l24.49-58h90.54l24.49 58H480zm-26.69 186L363 279.85 389.69 362zM272 320c-.25-.19-20.59-15.77-45.42-42.67 39.58-53.64 62-114.61 71.15-143.33H352V90H214V48h-44v42H32v44h219.25c-9.52 26.95-27.05 69.5-53.79 108.36-32.68-43.44-47.14-75.88-47.33-76.22L143 152l-38 22 6.87 13.86c.89 1.56 17.19 37.9 54.71 86.57.92 1.21 1.85 2.39 2.78 3.57-49.72 56.86-89.15 79.09-89.66 79.47L64 368l23 36 19.3-11.47c2.2-1.67 41.33-24 92-80.78 24.52 26.28 43.22 40.83 44.3 41.67L255 362z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="c2 "
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.536 9.146c-1.373 0-2.133 1.014-2.294 2.116h4.608c-.125-1.05-.867-2.115-2.314-2.115m-2.26 3.617c.235 1.156 1.193 1.97 2.532 1.97.725 0 1.77-.27 2.384-.914l1.175 1.35c-1.064 1.11-2.653 1.426-3.74 1.426-2.64 0-4.697-1.906-4.697-4.606 0-2.535 1.894-4.62 4.57-4.62 2.585 0 4.5 1.98 4.5 4.604v.766h-6.723v.023zm-6.487 3.83v-5.69c0-.976-.435-1.536-1.338-1.536-.814 0-1.355.585-1.717 1.007v6.24h-2.35v-5.7c0-.976-.415-1.532-1.318-1.532-.813 0-1.375.586-1.717 1.006v6.24H0V7.505h2.35v1.15c.4-.463 1.302-1.26 2.71-1.26 1.247 0 2.096.526 2.477 1.59.524-.761 1.5-1.59 2.91-1.59 1.7 0 2.69 1.01 2.69 2.963v6.24h-2.353l.005-.007z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="c2 "
      fill="none"
      focusable="false"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="c2 "
      fill="none"
      focusable="false"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <polyline
        points="16 12 12 8 8 12"
      />
      <line
        x1="12"
        x2="12"
        y1="16"
        y2="8"
      />
    </svg>
  </nav>
</div>
`;
